### PR TITLE
Adding install instruction for Ubuntu-CPU-Python

### PR DESCRIPTION
**Pull request by sandeep-krishnamurthy**
1. Adding new install page for MXNet. One page for all Programming language and Installation types with menus for users to choose.
2. Adding instructions to Ubuntu - Python - CPU : pip, virtualenv, docker, building from source.
3. This page is not directly linked from any mxnet.io page. Hence, will be still hidden. We will turn it on after completing instructions for Ubuntu GPU, Mac OS CPU and GPU.

Tested all the instructions on Ubuntu 14.04 and 16.04. Few screen shots attached.

@madjam - Can you please review and merge this? 
@zackchase  - Can you please provide your feedback? 
@lxn2  - Can you please review this changes and suggest if any changes are required to make it easy for your testing strategy?

screen shot 2017-04-18 at 3 32 56 pm
screen shot 2017-04-18 at 3 33 29 pm

### CODE REVIEW

**Comment by sandeep-krishnamurthy** [comment]
Since the page is hidden (no direct links from existing pages) and I have tested all the instructions work. I would like to get this merged and enable me to push other instructions (GPU, MacOS etc.) and review it at once for language and cosmetic.

It would be hard for reviewer to view this page in raw format for reviewing.

@madjam @zackchase thoughts?

File: docs/get_started/install.md [hunk 1]
```diff
@@ -0,0 +1,347 @@
...
+**Step 3**  Validate the installation by running a simple MXNet code described [here](#validate-mxnet-installation).
+
+</div>
+
+<div class="docker">
+<br/>
+
+Docker images with MXNet are available at [Docker Hub](https://hub.docker.com/r/mxnet/).
+
+**Step 1**  Install Docker on your machine following the [docker installation instructions](https://docs.docker.com/engine/installation/linux/ubuntu/#install-using-the-repository).
```
Comment (zackchase) [inline docs/get_started/install.md]: Install Docker on your machine *by* following the [docker installation instructions]

File: docs/get_started/install.md [hunk 2]
```diff
@@ -0,0 +1,347 @@
...
+
+*Note* - You can install Community Edition (CE) to get started with MXNet.
+
+**Step 2** [Optional] Post installation steps to manage Docker as a non-root user.
+
+Follow the 4 step instruction described in this [docker documentation](https://docs.docker.com/engine/installation/linux/linux-postinstall/#manage-docker-as-a-non-root-user) to allow managing docker containers without *sudo*.
```
Comment (zackchase) [inline docs/get_started/install.md]: Please change "4 step instruction described in this" 
to "four steps in this"

File: docs/get_started/install.md [hunk 3]
```diff
@@ -0,0 +1,347 @@
...
+You can list docker images to see if mxnet/python docker image pull was successful.
+
+```bash
+$ docker images # Use sudo if you skip Step 2
+
+REPOSITORY          TAG                 IMAGE ID            CREATED             SIZE
+mxnet/python        latest              00d026968b3c        3 weeks ago         1.41 GB
+```
+
+**Step 3** Validate the installation by running a simple MXNet code described [here](#validate-mxnet-installation).
```
Comment (zackchase) [inline docs/get_started/install.md]: same issue as above

File: docs/get_started/install.md [hunk 4]
```diff
@@ -0,0 +1,347 @@
...
+```
+
+**Step 2** Build the MXNet Python binding.
+
+```bash
+$ cd python
+$ sudo python setup.py install
+```
+
+**Step 3** Validate the installation by running a simple MXNet code described [here](#validate-mxnet-installation).
```
Comment (zackchase) [inline docs/get_started/install.md]: same issue as above

File: docs/get_started/install.md [hunk 5]
```diff
@@ -0,0 +1,347 @@
...
+
+<div class="virtualenv">
+
+Activate the virtualenv environment created for MXNet.
+
+```bash
+$ source ~/mxnet/bin/activate
+```
+
+After activating the environment, you should be seeing the prompt as below.
```
Comment (zackchase) [inline docs/get_started/install.md]: please change to "you should see the following prompt"

File: docs/get_started/install.md [hunk 6]
```diff
@@ -0,0 +1,347 @@
...
+
+Start the python terminal.
+
+```bash
+$ python
+```
+
+</div>
+
+Run a short mxnet python program to create a 2X3 identity matrix, multiply each element in the matrix by 2 followed by adding 1. We expect output to be a 2X3 matrix with all elements being 3.
```
Comment (zackchase) [inline docs/get_started/install.md]: lets's be consistent with how we refer to MXNet. Stick with _MXNet_.

File: docs/get_started/install.md [hunk 7]
```diff
@@ -0,0 +1,347 @@
...
+
+```python
+>>> exit()
+$
+```
+</div>
+
+<div class="virtualenv">
+
+Exit the Python terminal and Deactivate the virtualenv MXNet environment.
```
Comment (zackchase) [inline docs/get_started/install.md]: Is there a reason to capitalize Deactivate here? I think it should be "deactivate"

**Comment by sandeep-krishnamurthy** [comment]
@zackchase - Awesome. This review was very helpful.

I found a way to split this into multiple files and merge at build time. This splits will help in readability and testability.

I will address all these feedback and split the files and update the PR.

**Comment by sandeep-krishnamurthy** [comment]
Fixed comments from Zack.

@madjam - Can we get this reviewed and merged?

Thanks

**Pull request merged**